### DISCUSSION TITLE
Multiple sample type metadata templates

### DISCUSSION THREAD

**Discussion by mslarae13** [microbiomedata/issues#766] [General]
See:
- milestone 3.1 : https://github.com/microbiomedata/issues/issues/513
- https://github.com/microbiomedata/nmdc-server/issues/368

From the milestone, the goal is to allow people to make their own boutique metadata templates. From previous conversations, it's been discussed that this isn't really feasible on the submission portal. Different sample types have different requirements and validation isn't consistent.

However, another option is simply allowing for entering metadata from multiple sample types into 1 submission.
Right now, the NMDC submission portal has a single select for the extension and when doing sample submission to a UF, you can select JGI and EMSL and add metadata for each user facility.
Is it possible to do a similar action with the environmental extension? So users can submit say soil and plant metadata in 1 submission?

@pkalita-lbl @naglepuff @jeffbaumes thoughts?

@emileyfadrosh @cmungall @aclum

**Comment by aclum** [comment]
I like this proposed path forward.

**Comment by pkalita-lbl** [comment]
There are a few places in the submission portal codebase where we simply assume (because it's all that's been allowed so far) that a submission's sample data consists of exactly 1 environmental template and 0 or more user facility templates. We'd have to hunt those down and fix them up to allow 1 or more environmental templates in the sample data, but I don't think there are any major technical blockers there.

I do think it would be worth at least sketching out what this would look like from a user's perspective to make sure (a) we're not missing something and (b) we're all on the same page about the goal before diving into development.
